src/gengowatcher/web.py

```
@@ -254,6 +278,248 @@ async def cancel_current_job(self) -> bool:
+            return None
+        return self._build_file_entry(path)
+
+    @staticmethod
+    def _normalize_tier(
+        tier: str | None,
+        *,
+        word_count: int | None = None,
+        value: float | None = None,
+    ) -> str | None:
+        normalized = str(tier or "").strip().lower()
+        suffix = re.sub(r"[^A-Za-z0-9.]", "", suffix) or ".bin"
```

Ruff (0.15.9)

[error] Undefined name `suffix`

(F821)